lib/app/features/user/pages/profile_page/pages/settings/settings_modal.dart

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/components/screen_offset/screen_bottom_offset.dart';
+import 'package:ion/app/components/screen_offset/screen_side_offset.dart';
+import 'package:ion/app/constants/links.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/user/pages/switch_account_modal/components/action_button/action_button.dart';
+import 'package:ion/app/features/user/providers/app_info_provider.dart';
+import 'package:ion/app/router/app_routes.dart';
+import 'package:ion/app/router/components/navigation_app_bar/navigation_app_bar.dart';
+import 'package:ion/app/router/components/navigation_app_bar/navigation_close_button.dart';
+import 'package:ion/app/router/components/sheet_content/sheet_content.dart';
+import 'package:ion/app/services/browser/browser.dart';
+import 'package:ion/generated/assets.gen.dart';
+
+class SettingsModal extends ConsumerWidget {
+  const SettingsModal({required this.pubkey, super.key});
+
+  final String pubkey;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final iconColor = context.theme.appColors.primaryAccent;
+    final actions = <Widget>[
```

Comment:
remove` <Widget> `type

Reply:
Can you map the icons and labels (like in `FeedCategory` for example) and handle the list via `map()`?

Reply:
Made this only for `settings_modal`, because profile settings actions contains trailing widgets.

Reply:
Done.